lib/core/test/unit/Cardano/Wallet/Primitive/ModelSpec.hs

```diff
@@ -425,6 +438,202 @@ prop_availableUTxO makeProperty =
         shouldNotEvaluate fieldName = error $ unwords
             [fieldName, "was unexpectedly evaluated"]
 
+--------------------------------------------------------------------------------
+-- Change UTxO properties
+--------------------------------------------------------------------------------
+
+-- | Tests the 'changeUTxO' function with custom filter conditions to identify
+--   addresses that belong to the wallet.
+--
+-- This test divides all addresses into two disjoint sets:
+--
+--    - even-parity addresses:
+--      addresses with a pop count (Hamming weight) that is even.
+--
+--    - odd-parity addresses:
+--      addresses with a pop count (Hamming weight) that is odd.
+--
+-- With the above definitions in mind, this test uses the following filter
+-- conditions to categorize output addresses:
+--
+--    - 'Even' : matches addresses whose parity is even
+--    - 'Odd'  : matches addresses whose parity is odd
+--    - 'All'  : matches all addresses, regardless of their parity.
+--
+-- Each of the above filter conditions is represented with a custom 'IsOurs'
+-- instance that we provide to the 'changeUTxO' function.
+--
+-- Since a given address can either be even, or odd, but not both, it follows
+-- that the following results must be disjoint:
+--
+--    - 'changeUTxO pendingTxs Even'
+--    - 'changeUTxO pendingTxs Odd'
+--
+-- This test applies each of the above filter conditions to 'changeUTxO' and
+-- verifies that all entries match the condition that was provided.
+--
+prop_changeUTxO :: Property
+prop_changeUTxO =
+    forAllShrink (listOf genTxOutputs) (shrinkList shrinkTxOutputs) inner
+  where
+    inner :: [TxOutputs] -> Property
+    inner txOutputs =
```

Comment:
```suggestion
newtype Classify a = Classify (a -> Bool)

prop_changeUTxO :: Classify Address -> [TxOut] -> Property
prop_changeUTxO (Classify condition) txOutputs =
```

Reply:
@rvl Nice suggestion!

I've actually made a similar change in 0ffa43da16b9232aafcf613453709a279545cc74, which replaces the various types with a single `IsOursIf {condition :: a -> Bool}` newtype (with a single instance).